pandas/core/arrays/_arrow_utils.py

```diff
@@ -139,3 +145,78 @@ def to_pandas_dtype(self):
 # register the type with a dummy instance
 _interval_type = ArrowIntervalType(pyarrow.int64(), "left")
 pyarrow.register_extension_type(_interval_type)
+
+
+ArrowExtensionArrayT = TypeVar("ArrowExtensionArrayT", bound="ArrowExtensionArray")
+
+
+class ArrowExtensionArray(ExtensionArray):
+    """
+    Base class for ExtensionArray backed by Arrow array.
+    """
+
+    _data: pyarrow.ChunkedArray
+
+    def __init__(self, values: pyarrow.ChunkedArray):
+        raise NotImplementedError
+
+    def __arrow_array__(self, type=None):
+        """Convert myself to a pyarrow Array or ChunkedArray."""
+        return self._data
+
+    @property
+    def nbytes(self) -> int:
+        """
+        The number of bytes needed to store this object in memory.
+        """
+        return self._data.nbytes
+
+    def __len__(self) -> int:
+        """
+        Length of this array.
+
+        Returns
+        -------
+        length : int
+        """
+        return len(self._data)
+
+    def isna(self) -> npt.NDArray[np.bool_]:
+        """
+        Boolean NumPy array indicating if each value is missing.
+
+        This should return a 1-D array the same length as 'self'.
+        """
+        # TODO: Implement .to_numpy for ChunkedArray
+        return self._data.is_null().to_pandas().values
```

Comment:
pa.Array/pa.ChunkedArray's should have `to_numpy()`?

https://arrow.apache.org/docs/python/generated/pyarrow.ChunkedArray.html#pyarrow.ChunkedArray.to_numpy

Reply:
Good question.  I cut/pasted the implementation in ArrowStringArray following a conversation with @jorisvandenbossche.  Maybe the comment here is out of date?

Reply:
ChunkedArray.to_numpy was added in pyarrow 2.0, so depending on the minimum target version, you can use it or not